src/elements/content-explorer/ContentExplorer.js

```diff
@@ -844,18 +845,29 @@ class ContentExplorer extends Component<Props, State> {
         const fileAPI = this.api.getFileAPI(false);
         const newCollection: Collection = { ...collection };
         const selectedId = selectedItem ? selectedItem.id : null;
-        const thumbnails = await Promise.all(items.map(item => fileAPI.getThumbnailUrl(item)));
         let newSelectedItem: ?BoxItem;
 
-        newCollection.items = items.map((obj, index) => {
-            const isSelected = obj.id === selectedId;
-            const currentItem = isSelected ? selectedItem : obj;
+        const itemThumbnails = await Promise.all(
+            items.map(item => {
+                return item.type === TYPE_FILE ? fileAPI.getThumbnailUrl(item) : null;
+            }),
+        );
+
+        newCollection.items = items.map((item, index) => {
+            const isSelected = item.id === selectedId;
+            const currentItem = isSelected ? selectedItem : item;
+            const thumbnailUrl = itemThumbnails[index];
+
             const newItem = {
                 ...currentItem,
                 selected: isSelected,
-                thumbnailUrl: thumbnails[index],
+                thumbnailUrl,
             };
 
+            if (item.type === TYPE_FILE && thumbnailUrl && !isThumbnailReady(newItem)) {
```

Comment:
Seems like `updateCollection` gets called in various ways, after view change (search, recents, folder) as well as page operations (select, sharedlink). Is it possible that the thumbnail generation could get duplicated for instance after each check selection user interaction?

Reply:
Just tested these flows. `attemptThumbnailGeneration()` only gets called when `isThumbnailReady()` returns false, and that result only occurs if the representation status is neither "viewable" nor "success." So, thumbnails that have already been generated will not be generated again.

Reply:
👍 thanks for checking